core/src/main/java/software/amazon/awssdk/auth/ProfileCredentialsProvider.java

```diff
@@ -70,7 +71,11 @@ public static Builder builder() {
 
     @Override
     protected AwsCredentials loadCredentials() {
-        return profilesConfigFile == null ? null : profilesConfigFile.getCredentials(profileName);
+        if (profilesConfigFile == null) {
+            throw new SdkClientException("Unable to find profile configuration file.");
```

Comment:
Is it possible to specify how they can fix this? It might be tricky with the location provider being a separate abstraction.